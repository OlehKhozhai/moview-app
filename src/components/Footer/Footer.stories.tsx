import React from 'react';
import { Meta } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';

import Footer from './index';

export default {
  component: Footer,
  title: 'components/Footer',
} as Meta;

export const FooterComponent = () => (
  <MemoryRouter>
    <div style={{ width: 1440 }}>
      <Footer />
    </div>
  </MemoryRouter>
);
