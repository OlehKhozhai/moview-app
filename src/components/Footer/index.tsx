import React from 'react';

import Logo from 'components/_common/Logo';
import styles from './styles.module.scss';

const Footer: React.FC = () => {
  return (
    <footer className={styles.root}>
      <Logo />
    </footer>
  );
};

export default React.memo(Footer);
