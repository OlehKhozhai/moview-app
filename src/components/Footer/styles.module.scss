@import 'assets/styles/mixins/colors';

.root {
  padding: 20px 0;
  text-align: center;
  background-color: $primary-dark-bg;
}
