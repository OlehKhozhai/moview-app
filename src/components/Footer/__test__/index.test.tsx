import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';

import Footer from '../index';

describe('Footer component', () => {
  test('should render Footer component by default', () => {
    const component = create(
      <MemoryRouter initialEntries={['/']}>
        <Footer />
      </MemoryRouter>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
