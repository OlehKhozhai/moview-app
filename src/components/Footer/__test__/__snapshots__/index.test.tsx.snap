// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should render Footer component by default 1`] = `
<footer
  className="root"
>
  <a
    className="root"
    href="/"
    onClick={[Function]}
  >
    <span
      className="bold"
    >
      netflix
    </span>
    roulete
  </a>
</footer>
`;
